server/graphql/v2/input/AccountInput.js

```diff
@@ -0,0 +1,45 @@
+import { GraphQLInt, GraphQLString, GraphQLInputObjectType } from 'graphql';
+
+import models from '../../../models';
+import { idDecode } from '../identifiers';
+import { NotFound } from '../../errors';
+
+const AccountInput = new GraphQLInputObjectType({
+  name: 'AccountInput',
+  fields: () => ({
+    id: {
+      type: GraphQLString,
+      description: 'The public id identifying the account (ie: dgm9bnk8-0437xqry-ejpvzeol-jdayw5re)',
+    },
+    legacyId: {
+      type: GraphQLInt,
+      description: 'The internal id the account (ie: 580)',
+      deprecationReason: '2020-01-01: should only be used during the transition to GraphQL API v2.',
+    },
+    slug: {
+      type: GraphQLString,
+      description: 'The slug identifying the account (ie: babel for https://opencollective.com/babel)',
+    },
+  }),
+});
+
+const fetchAccountWithInput = async (input, options) => {
+  let collective;
+  if (input.id) {
+    const id = idDecode(input.id, 'account');
+    collective = await models.Collective.findByPk(id);
+  } else if (input.legacyId) {
+    collective = await models.Collective.findByPk(input.legacyId);
+  } else if (input.slug) {
+    const slug = input.slug.toLowerCase();
+    collective = await models.Collective.findBySlug(slug);
+  } else {
+    throw new Error('Please provide an id or a slug');
+  }
+  if (!collective && options && options.throwIfMissing) {
+    throw new NotFound({ message: 'Account Not Found' });
+  }
+  return collective;
+};
```

Comment:
Can we use the GraphQL loaders in there to make it safe to use this in sub-fields?